### DISCUSSION TITLE
Cruiser settings

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#31] [Q&A] [answered]
One of the many strong features of the Cruiser app is its _many configuration options_.

We need to reorganize the Cruiser settings and separate them into different screens.

So they don't look endless / complex to users and group them better for new features.
e.g. [Screen margins](https://github.com/devemux86/cruiser/discussions/24)

**Comment by DolomitesTom** [comment]
> One of the many strong features of the Cruiser app is its many configuration options.

absolutely, perfect. 

> We need to reorganize the Cruiser settings and separate them into different screens.

I am used to it in the meantime. What about 2 modes? Standard and Advanced? Or beginner and expert?

**Reply by Leeb64** [reply]
I'm used to it, but it could - maybe - be better.
A button from "Routing" directly in the routing settings to select the routing service, for example....

**Comment by devemux86** [comment]
As a first step we can use separate screens for the already grouped settings.
So they do not appear complicated in endless lists, e.g. navigation settings.
(structure remains the same)

Then we could indeed put some settings into advanced groups.

**Reply by DolomitesTom** [reply]
Yes, looks good

**Comment by devemux86** [comment]
This way we can include extensive options per routing service.
e.g. handle the multiple BRouter profiles within Cruiser settings.

**Reply by Leeb64** [reply]
Good Idea !

**Comment by airald** [comment]
I find the settings are so good sorted in different sections. Of course, all can be better.

Maybe could be usefull a "search" function to find the desired option. If you don't know all the options, you should have a good user manual or just navigate investigating different options.

**Reply by devemux86** [reply]
> Maybe could be useful a "search" function to find the desired option.

Indeed that would be helpful.

**Reply by moving-bits** [reply]
If you want to take a look: c:geo supports searching in it's preferences, just tap on the search icon and start typing:

|pref start screen|search activated|some keys later|
|---|---|---|
|![image](https://user-images.githubusercontent.com/3754370/216844668-b530dbc8-093b-424c-9877-76424d1c3236.png)|![image](https://user-images.githubusercontent.com/3754370/216844676-51a729e4-1110-4e55-8ed8-8bdb9cb07df3.png)|![image](https://user-images.githubusercontent.com/3754370/216844682-471f2692-f355-4676-ad35-707e513bc3b6.png)|

See `settings/SettingsActivity.buildSearchIndex()` and the infrastructure prepared in `settings/BasePreferenceFragment`

**Comment by devemux86** [comment]
_**Simple by default - Powerful when needed**_

---

Cruiser will have its many configuration options organized in groups and advanced settings:

<img src="https://user-images.githubusercontent.com/3484020/218750095-f922537a-26c6-4e76-9072-7c5d11f424f9.jpg" width="50%"/><img src="https://user-images.githubusercontent.com/3484020/218750105-f7d29e27-d38d-4525-90b9-6c17377593c8.jpg" width="50%"/>

**Comment by devemux86** [comment]
**Search in settings** will be available in the next Cruiser:

<img src="https://user-images.githubusercontent.com/3484020/219051513-e73e906b-5b60-41d5-847a-dd23c5961b80.jpg" width="50%"/><img src="https://user-images.githubusercontent.com/3484020/219367931-d655fc97-b7a7-43bc-af46-398b98d47084.jpg" width="50%"/>

**Reply by DolomitesTom** [reply]
Release it please in order to be able to test it 😁

**Reply by devemux86** [reply]
Let's make **search in settings** more useful / beautiful:
- Color search will _show colors in results_
(see updated images above in https://github.com/devemux86/cruiser/discussions/31#discussioncomment-4983914)

**Comment by devemux86** [comment, marked as answer]
[Cruiser 3.0.9](https://play.google.com/store/apps/details?id=gr.talent.cruiser) is available with:

- Settings new design: https://github.com/devemux86/cruiser/discussions/31#discussioncomment-4971075
- **Search** in settings: https://github.com/devemux86/cruiser/discussions/31#discussioncomment-4983914

**Reply by WolfgangZ42** [reply]
My compliments - the settings are much clearer and the search function certainly helps.

**Comment by DolomitesTom** [comment]
really good

**Comment by devemux86** [comment]
Settings in the search results display their values (even [colors](https://github.com/devemux86/cruiser/discussions/31#discussioncomment-4983914)), except for the on/off settings.

This will be available in the next Cruiser update (via check marks),
so you can easily see which settings are enabled:

<img src="https://user-images.githubusercontent.com/3484020/224382229-2a420e68-140a-4c7d-a1f1-4e547e0fa537.jpg" width="50%"/>

**Comment by devemux86** [comment]
In the next Cruiser update, all settings that require a restart of the app will be applied automatically without user intervention.

**Comment by devemux86** [comment]
> Settings in the search results display their values (even https://github.com/devemux86/cruiser/discussions/31#discussioncomment-4983914), except for the on/off settings.
> This will be available in the next Cruiser update (via check marks), so you can easily see which settings are enabled.

> In the next Cruiser update, all settings that require a restart of the app will be applied automatically without user intervention.

Both are implemented in [Cruiser 3.0.13](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Comment by devemux86** [comment]
Next Cruiser will show the panel icons in the navigation display settings, so you can identify the panels:

<img src="https://user-images.githubusercontent.com/3484020/233592727-02fa70a6-89e2-457c-903f-6594d0a35dd6.jpg" width="50%"/>

**Reply by ramondevic** [reply]
But you won't remove these options at the bottom of the screen, I guess.

![imatge](https://user-images.githubusercontent.com/122691401/233075954-1ce4a513-5d65-445c-b339-b2abcf799990.png)

**Reply by devemux86** [reply]
@ramondevic all settings remain, they did not fit in the screenshot. :slightly_smiling_face:
